js/import/util/metadata.js

```diff
@@ -0,0 +1,162 @@
+import alert from '../../shared/alert.js';
+import { saveImporterSectionsMapping } from '../../sections-mapping/utils.ui.js';
+import { createElement } from '../../shared/utils.js';
+
+const METADATA_EDITOR = document.getElementById('metadata-editor');
+const METADATA_EDITOR_SECTIONS = document.getElementById('metadata-editor-rows');
+const ADD_METADATA_BUTTON = document.getElementById('metadata-editor-add');
+
+// Handler for deleting a row for this scope (copied until `mappingData` is centralized).
+let getMetadataRowDeleteButton;
+
+/**
+ * Either find the matching mapping and update it, create a brand new one or indicate that there
+ * is a duplicate.
+ * @param event
+ * @param mappingData
+ * @param originalURL
+ */
+const updateMetadataMapping = (event, mappingData, originalURL) => {
+  if (event.target.value || event.target.value === '') {
+    const metadataFields = ['name', 'value', 'condition'];
```

Comment:
FUTURE: Adding `'name', 'value', 'condition'` properties to a mapping object is not `metadata` specific but block config specific. Any block that requires a block config could use the same mapping data. Eventually, I can see separating out the addition of `'name', 'value', 'condition'` properties to an `addBlockConfigMapping` function so that it could be resued across multiple blocks.